quic/s2n-quic-platform/src/message/msg/handle.rs

```diff
@@ -64,15 +65,43 @@ impl path::Handle for Handle {
     }
 
     #[inline]
-    fn eq(&self, other: &Self) -> bool {
-        let mut eq = true;
+    fn set_local_address(&mut self, addr: LocalAddress) {
+        self.local_address = addr;
+    }
+
+    #[inline]
+    fn unmapped_eq(&self, other: &Self) -> bool {
+        ensure!(
+            self.remote_address.unmapped_eq(&other.remote_address),
+            false
+        );
 
         // only compare local addresses if the OS returns them
-        if features::pktinfo::IS_SUPPORTED {
-            eq &= self.local_address.eq(&other.local_address);
+        if !features::pktinfo::IS_SUPPORTED {
+            return true;
+        }
```

Comment:
can you use `ensure!` here?
```
ensure!(features::pktinfo::IS_SUPPORTED, true);
```

Reply:
Yeah I was going to but I thought it might be less readable to return false there? I can change it if you disagree

Reply:
why return false?

Reply:
Sorry I meant `true` 😁. I'm responding on my phone

Reply:
I'm ok either way